pepper-apis/src/main/java/org/broadinstitute/ddp/json/CreateTemporaryUserPayload.java

```diff
@@ -10,11 +10,20 @@
     @SerializedName("auth0ClientId")
     private String auth0ClientId;
 
-    public CreateTemporaryUserPayload(String auth0ClientId) {
+    @NotBlank
+    @SerializedName("auth0Domain")
+    private String auth0Domain;
+
+    public CreateTemporaryUserPayload(String auth0ClientId, String auth0Domain) {
         this.auth0ClientId = auth0ClientId;
+        this.auth0Domain = auth0Domain;
```

Comment:
In light of the recent discussion around  breaking API changes, I think we can make things easier on clients by looking up the `auth0Domain` based on `auth0ClientId` and erroring out if there are 0 or > 1 rows returned.  The change is still breaking, but there would be a grace period where clients are unlikely to break until we see duplicate client ids appear.

I believe we already include this value as `auth0Audience` on the angular side, so this should be a small change for angular.  Not sure about mobile app.

CC @bskinner @fat23cat @MocanaAtBroad

Reply:
This sounds like a nice idea for backwards-compatibility!

Reply:
That's an excellent idea- I'm behind it!

Reply:
FYI I'm tracking API breaking changes here: https://broadinstitute.atlassian.net/browse/DDP-4471.

Reply:
I implemented the approach suggested by @zyme: if a domain is missing, we check if the clientId is unique. If it is, we proceed or else fail. If the domain is present, we use it. From the client's point of view, there is no difference until we start getting clientId clashes.